### DISCUSSION TITLE
dylib corruption.

### DISCUSSION THREAD

**Discussion by colleeseum** [Homebrew/discussions#1223] [Everyday usage] [answered]
Hi, we have a formula (https://github.com/stardog-union/homebrew-tap/blob/master/Formula/stardog.rb)

Which for the most part works well, but the dylib are getting corrupted and therefore the application does not to work. We validated this my copying back the original, at which point application will work as expected

I have track the issue to keg.rb. I executed the change manually on the CLI and I do get this warning
`/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: libStarrocks.dylib`

As mentioned, everything works without apply this change, however even after groking the code or the documentation, I have not found a way to skip performing this action. 

Anyone has any idea. ?

**Comment by SMillerDev** [comment]
> I have track the issue to keg.rb. I executed the change manually on the CLI and I do get this warning

What change?

**Comment by colleeseum** [comment]
```
install_name_tool -id /usr/local/opt/stardog/lib/libvw_jni.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: libvw_jni.dylib
```

this is performed by the following method

 ```
 def change_dylib_id(id, file)
    return if file.dylib_id == id

    @require_relocation = true
    odebug "Changing dylib ID of #{file}\n  from #{file.dylib_id}\n    to #{id}"
    MachO::Tools.change_dylib_id(file, id, strict: false)
    apply_ad_hoc_signature(file)
  rescue MachO::MachOError
    onoe <<~EOS
      Failed changing dylib ID of #{file}
        from #{file.dylib_id}
          to #{id}
    EOS
    raise
  end
```

**Comment by carlocab** [comment, marked as answer]
I guess part of the issue is that you're installing binary-only formulae, which isn't really the intended use case here. (See the [docs](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae).)

I don't think there's a way to skip this. You can either try to use the formula to build from source or build an installer and make this a cask instead.

**Comment by colleeseum** [comment]
Thanks, while it was not the answer I was hoping for (more work for us), it was very helpful.

**Reply by carlocab** [reply]
Sure. If you go via the build-from-source-in-the-formula route, you can still use the formula to install pre-built packages by adding a bottle block to the formula.

Some of these docs might help

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/
